### ISSUE TITLE
Python 3.10 breaks qutip install

### ISSUE THREAD

**Issue by quantshah** [qutip/qutip#1732]
### Bug Description

In a fresh conda environment with python 3.10, qutip installation from pip fails.

It seems that this was due to people jumping to Python 3.10 too early https://github.com/numpy/numpy/issues/20039. Since October this issue is resolved and we can install Numpy from pip with Python 3.10.

However, trying to install qutip with Python 3.10 still leads to this error. Even if I had Numpy (numpy==1.21.4) already installed, qutip tries to build Numpy again. I cannot install numpy 1.20.0 with Python 3.10 and so downgrading my Python to 3.9 is the only solution for now.

 @hodgestar @jakelishman We should try to find a fix for this ASAP . Any thoughts on this?

### Code to Reproduce the Bug

```shell
conda create -n py310 python=3.10
conda activate py310
pip install qutip
```

### Code Output

```shell
ERROR: Failed building wheel for numpy
  Failed to build numpy
  ERROR: Could not build wheels for numpy, which is required to install pyproject.toml-based projects
```
```


### Expected Behaviour

Qutip should have got installed after installing the dependencies

### Your Environment

```shell
N/A.
```

**Comment by jakelishman** [comment]
It's related to #1720 - only Numpy 1.21 has cpy310 wheels, but the build system currently requires Numpy <1.20 to build, because when we build QuTiP wheels for general consumption, we need to use Numpy pre-1.20 in order to have runtime support for a range of Numpys.

As a quick fix for this issue with 3.10 only, you should be able to changes these lines in `setup.cfg`:
```
setup_requires =
    numpy>=1.16.6,<1.20
    scipy>=1.0
    cython>=0.29.20
    packaging
```
to
```
setup_requires =
    numpy>=1.16.6,<1.20 ; python_version<"3.10"
    numpy>=1.21 ; python_version>="3.10"
    scipy>=1.0
    cython>=0.29.20
    packaging
```
(or something like that - I forget the exact syntax).  That should let you build on 3.10 (where pre-1.20 Numpy versions aren't an issue), but it doesn't solve the general problem of #1720.

**Comment by hodgestar** [comment]
@quantshah I've merged #1747. That doesn't provide pre-built wheels or conda-forge packages for 3.10, but building for 3.10 locally should work now. Would you mind trying it out when you get a chance?
